### PR TITLE
Adding methods responsible for moving match within one digit and in w…

### PR DESCRIPTION
**Pull request by Frexor**
Save generated equations in one array.

### CODE REVIEW

File: src/EquationGenerator.java [hunk 1]
```diff
@@ -62,4 +47,96 @@ void generateCorrectEquations() {
...
+        canTakeFrom['='] = new char[]{};
+
+        int k = 0;
+        for (int i = 0; i < 5; i++) {
+            for (int j = 0; j < canTakeFrom[equation.charAt(i)].length; j++) {
+                char temp = equation.charAt(i);
+                equationChar[i] = canTakeFrom[equation.charAt(i)][j];
+                String equationWithoutMatch = new String(equationChar);
+                String[] equationsWithAddedMatch = addMatchToCharacter(equationWithoutMatch, i);
+                for (String equationWithAddedMatch : equationsWithAddedMatch) {
```
Comment (piekaa) [inline src/EquationGenerator.java:106]: System.arraycopy could have been used, but it's ok

File: src/EquationGenerator.java [hunk 2]
```diff
@@ -62,4 +47,96 @@ void generateCorrectEquations() {
...
+            }
+        }
+        String[] results = new String[k];
+        System.arraycopy(tempResults, 0, results, 0, k);
+
+        return results;
+
+    }
+
+    String[] allPossibleMatchMoves(String equation) {
```
Comment (piekaa) [inline src/EquationGenerator.java]: This method should be above correctEquationToQuizzes and changeMatchPositionBetweenCharacters

File: src/EquationGenerator.java [hunk 3]
```diff
@@ -62,4 +47,96 @@ void generateCorrectEquations() {
...
+        return results;
+
+    }
+
+    String[] addMatchToCharacter(String equation, int position) {
+
+        char[] equationChar = equation.toCharArray();
+        String[] tempResults = new String[100];
+
+        char[][] canAddTo = new char[255][];
```
Comment (piekaa) [inline src/EquationGenerator.java]: This should be class field, fill it with values in constructor. There is no point in creating new array in each invocation. It's just waste of time and memory. Same applies to canTakeFrom  and canChangePosition  arrays.

**Review by piekaa — CHANGES_REQUESTED** [review]
Small changes required.

File: src/EquationGenerator.java (hunk 2 above)
Comment (Frexor) [reply src/EquationGenerator.java]: Done

File: src/EquationGenerator.java (hunk 3 above)
Comment (Frexor) [reply src/EquationGenerator.java]: Done

**Review by piekaa — APPROVED** [review]
Good

**Pull request merged**